src/core/api/cart/add-product.ts

```diff
@@ -0,0 +1,38 @@
+// import { withAnonymousSessionFlow } from '@/core/api/middlewere/anonymous-session-flow.ts';
+import { MyCartAddLineItemAction, Cart, ClientResponse } from '@commercetools/platform-sdk';
+import { ACTIONS } from '@/constants/constants.ts';
+import { getApiClientForUser } from '@/core/services/api-client-options.ts';
+
+export const addProductToCart = async (
+  cartId: string,
+  version: number,
+  productId: string,
+  variantId: number,
+  userToken?: string | null,
+  quantity = 1,
+): Promise<Cart> => {
+  try {
+    const cartData: MyCartAddLineItemAction = {
```

Comment:
How about using an object as an argument with spread?

```suggestion
interface IProductParams {
  cartId: string;
  version: number;
  productId: string;
  variantId: number;
  userToken?: string | null;
}

export const addProductToCart = async (productParams: IProductParams, quantity = 1): Promise<Cart> => {
  const { cartId, version, productId, variantId, userToken } = productParams;
  try {
    const cartData: MyCartAddLineItemAction = {
```